DT-436 adding tags to aws resources

christopher-comet opened the pull request:
https://comet-ml.atlassian.net/browse/DT-436

This also includes edits to make the tfvars ready to go with defaults and updates the redis version.

christopher-comet commented on terraform.tfvars:
```diff
@@ -53,24 +62,24 @@ comet_private_subnets = ["subnet-012345abcdefghijkl", "subnet-012345abcdefghijkl
 
 #### comet_eks ####
-#
+# eks_aws_cloudwatch_metrics = false
```
In my testing toggling this to false (uncommented) will prevent the creation of the "amazon-cloudwatch" namespace.

darenjacobs reviewed (approved):
LGTM

christopher-comet commented on variables.tf:
```diff
@@ -375,3 +375,33 @@ variable "single_nat_gateway" {
+  description = "A map of tags to apply to resources"
+  type        = map(string)
+
+  default = {
```
For customers:
Defaults should just be an empty dictionary with a comment that tells users to place their default set of tags here.

We also want our set of internal tags. Including:
`Environment = var.environment_tag`

christopher-comet commented on variables.tf:
```diff
@@ -375,3 +375,33 @@ variable "single_nat_gateway" {
+  type        = map(string)
+
+  default = {
+    Environment    = ""
+    Owner          = ""
+    DeployedBy     = ""
+    Terraform      = "true"
+    Customer       = ""
+    TTL            = ""
+    Product        = ""
```
```suggestion
    Environment    = var.environment_tag
    Terraform      = "true"
```

christopher-comet commented on variables.tf:
```diff
@@ -375,3 +375,33 @@ variable "single_nat_gateway" {
+variable "common_tags" {
+  description = "A map of tags to apply to resources"
+  type        = map(string)
+
```
```suggestion

# Place your dictionary of customized tags for AWS resources here
# eg.
# Owner = "devops"
# DeployedBy = "firstName-lastName"
# TTL = ""
# Customer = ""
```

christopher-comet commented:
Ready for review. @darenjacobs @liyaka

liyaka commented on main.tf:
```diff
@@ -5,10 +5,13 @@ data "aws_eks_cluster_auth" "this" {
+  merged_tags = merge(
+    var.common_tags,
+    {
+      Environment = var.environment
```
We should use a different variable for the Environment tag, since var.environment is used in many other places, and for our managed environments we will want to use a different value, like 'deployment-production'

liyaka commented on modules/comet_ec2/variables.tf:
```diff
@@ -68,4 +68,9 @@ variable "comet_ec2_s3_iam_policy" {
+variable "common_tags" {
+  type        = map(string)
+  description = "A map of common tags"
+}
```
you have to set the default:
```
default = {}
```

liyaka commented on modules/comet_eks/variables.tf:
```diff
@@ -113,4 +113,9 @@ variable "eks_airflow_instance_type" {
+variable "common_tags" {
+  type        = map(string)
+  description = "A map of common tags"
+}
```
you have to set the default:
```
default = {}
```

liyaka commented on modules/comet_elasticache/variables.tf:
```diff
@@ -52,4 +52,9 @@ variable "elasticache_auth_token" {
+variable "common_tags" {
+  type        = map(string)
+  description = "A map of common tags"
+}
```
you have to set the default:
```
default = {}
```

liyaka commented on modules/comet_rds/variables.tf:
```diff
@@ -72,3 +72,8 @@ variable "rds_root_password" {
+variable "common_tags" {
+  type        = map(string)
+  description = "A map of common tags"
+}
```
you have to set the default:
```
default = {}
```

liyaka commented on modules/comet_s3/variables.tf:
```diff
@@ -16,4 +16,9 @@ variable "s3_force_destroy" {
+variable "common_tags" {
+  type        = map(string)
+  description = "A map of common tags"
+}
```
you have to set the default:
```
default = {}
```

liyaka commented on modules/comet_vpc/variables.tf:
```diff
@@ -11,4 +11,9 @@ variable "eks_enabled" {
+variable "common_tags" {
+  type        = map(string)
+  description = "A map of common tags"
+}
```
you have to set the default:
```
default = {}
```

liyaka commented on terraform.tfvars:
```diff
@@ -1,29 +1,41 @@
+#   DeployedBy     = "Devops"
+#   TTL            = "2025-01-01 12:00:00"
+#   Customer       = "Model-Ops"
+#   EnvironmentTag = "test"
```
I suggest to remove this example of EnvironmentTag, it is confusing, and we already use Environment tag

liyaka commented on variables.tf:
```diff
@@ -375,3 +375,8 @@ variable "single_nat_gateway" {
+variable "common_tags" {
+  description = "A map of tags to apply to resources"
+  type        = map(string)
+}
```
you have to set the default:
```
default = {}
```

liyaka commented on providers.tf:
```diff
@@ -1,5 +1,9 @@
 provider "aws" {
   region = var.region
+
+  default_tags {
```
provider default tag should include
Terraform = "true" and Environment tags

christopher-comet commented on providers.tf:
```diff
@@ -1,5 +1,15 @@
+    tags = merge(
+      {
+        Terraform   = "true"
+        Environment = var.environment
```
```suggestion
        Environment = var.environment_tag
```

christopher-comet commented:
There are a couple of modules wherein the resource explicitly sets tags.
eg. in `terraform-aws-comet/modules/comet_s3/main.tf`
```
resource "aws_s3_bucket" "comet_airflow_bucket" {
  count = var.enable_mpm_infra ? 1 : 0

  bucket = "comet-airflow-${local.suffix}"

  force_destroy = var.s3_force_destroy

  tags = {
    Name = "comet-airflow-${local.suffix}"
  }
}
```
In this situation the S3 bucket will only get the Name tag, and it will NOT inherit common_tags automatically because the tags block inside aws_s3_bucket is explicitly defined, and default_tags from providers.tf does not override manually set tags.

Therefore for these cases we need to merge the common_tags.

christopher-comet commented:
✅ Summary of Terraform Changes in This PR
🔁 Provider Updates

    Bumped versions for the following providers:

        aws → 5.97.0

        helm → 2.17.0

        kubernetes → 2.36.0

        Others: tls, time, cloudinit, and added null

    Why: Modernization, bug fixes, and compatibility with new module versions.

EKS Module & Provider Refactor

    Migrated to terraform-aws-modules/eks v20.33.1

    Set authentication_mode = "API_AND_CONFIG_MAP" for streamlined auth

    Replaced deprecated aws_eks_cluster_auth with output-based kubeconfig logic

    Moved helm.eks and kubernetes.eks provider definitions into eks_providers.tf

    Added explicit providers = { ... } block for comet_eks

Enables one-stage apply without Helm/Kubernetes provider race conditions.
 Unified Tagging Across Modules

    Replaced hardcoded tags = { Terraform = "true", Environment = var.environment }

    Now using common_tags passed into all modules

    Centralized in local.all_tags and merged with per-resource tags

 Improves tag consistency across all infrastructure components.
 Module Improvements

    Updated modules for:

        comet_eks (with conditional merge for node groups)

        comet_ec2, comet_vpc, comet_elasticache, comet_rds, comet_s3

    Added missing common_tags variables and passed them to resources

 Output Cleanup

    Removed obsolete comet_eks_token output (no longer available via token-based auth)

    Updated comet_eks_cert and comet_eks_endpoint to match new module structure

 IRSA + Addons

    Added IRSA role creation for EBS CSI driver

    Integrated aws-ia/eks-blueprints-addons module to install:

        CoreDNS, kube-proxy, vpc-cni, aws-ebs-csi-driver

        AWS Load Balancer Controller (with Helm release)
